x-pack/plugins/encrypted_saved_objects/server/saved_objects/index.ts

```diff
@@ -96,6 +106,38 @@ export function setupSavedObjects({
           )) as T,
         };
       },
+
+      createPointInTimeFinderAsInternalUser: async <T = unknown, A = unknown>(
+        findOptions: SavedObjectsCreatePointInTimeFinderOptions,
+        dependencies?: SavedObjectsCreatePointInTimeFinderDependencies
```

Comment:
__question:__ I tried to find any info on the `dependencies` parameter to understand the implications of supporting it in ESO, but couldn't. @elastic/kibana-core what `SavedObjectsCreatePointInTimeFinderDependencies` in `createPointInTimeFinder` API is for? I'm wondering if it provides any reasonable benefits to ESO consumers or we can ignore it if it's more an internal Core thing.

Reply:
I can see, here is only one property available

```typescript
export interface SavedObjectsCreatePointInTimeFinderDependencies {
  client: Pick<SavedObjectsClientContract, 'find' | 'openPointInTimeForType' | 'closePointInTime'>;
}
```
If I'm not mistaken, looking at code, SO client used here:

```
        const [internalRepository, typeRegistry] = await internalRepositoryAndTypeRegistryPromise;
        const finder = internalRepository.createPointInTimeFinder<T, A>(findOptions, dependencies);
```

is wrapped by EncryptedSavedObjectsClientWrapper? 

which at the same time implemented as:

```typescript
  public createPointInTimeFinder<T = unknown, A = unknown>(
    findOptions: SavedObjectsCreatePointInTimeFinderOptions,
    dependencies?: SavedObjectsCreatePointInTimeFinderDependencies
  ) {
    return this.options.baseClient.createPointInTimeFinder<T, A>(findOptions, {
      client: this,
      // Include dependencies last so that subsequent SO client wrappers have their settings applied.
      ...dependencies,
    });
  }

```

Looks like, this is an ability to add some wrapper on top of SO client.